### ISSUE TITLE
DBUnit/JPA Conflict Since 1.3.0.M5

### ISSUE THREAD

**Issue by donnyhallman** [spring-projects/spring-boot#4128]
I notice jumping from 1.3.0.M4 to M5 (including BUILD-SNAPSHOT) I get the following with JPA configured application at least when using DBUnit:

```java
org.dbunit.dataset.NoSuchColumnException: xxxxxxx.ID -  (Non-uppercase input column: id) in ColumnNameToIndexes cache map. Note that the map's column names are NOT case sensitive.
```

Using DBUnit 2.5.1 with H2 in memory database. I'm using Liquibase at the moment for setting up the schema for test setup. Also, I have the following configurations:

> spring.jpa.show-sql=true
> spring.jpa.hibernate.ddl-auto=none

Reverting to M4 resolves this conflict for now.

**Comment by snicoll** [comment]
What makes you think it's a Spring Boot issue exactly? It looks like your database schema used to be created with upper case and is not anymore.

**Comment by donnyhallman** [comment]
Nothing particular changed with the schema that I'm aware of. However, prior to adding **spring.jpa.hibernate.ddl-auto=none**, I was having an issue unrelated to the spring boot version which was caused by hibernate auto generating the schema, which was not desired. 

The exception I mentioned is often attributed to [MySQL by DBUnit](http://dbunit.sourceforge.net/faq.html#NoSuchColumnException), which normally would be the database used in most environments for my project except in this case which is for integration testing, in which H2 is used (properties overwritten using test configuration).

It may be that there was a change to Entity Management/Scan which may be affecting this issue. I won't discount the possibility that I'm missing a configuration that wasn't exposed prior to M5. However, I will note that the Liquibase configurations, JPA annotations, and DbUnit files don't change when I go between versions.

All of that being said, give me an opportunity to verify that I see differences between testing and running application and I'll follow up.

**Comment by donnyhallman** [comment]
Sorry for the delayed follow up:

**Short Answer** -  H2 v1.4.188 (default for boot M5+) = problem; 1.4.189 = solved. Consider H2 1.4.189 for 1.3.0 RC.

**Long Answer**:
I did some looking, there were no problems when I deployed my application against MySQL. Even, in boot 1.3.0M4 if I use H2 1.4.188 I get the same problem. If I use H2 1.4.189 (and the older 1.4.187) it works fine. This proves true for M5 and BUILD-SNAPSHOT. Choose the more recent H2 version perhaps  should be considered for 1.3.0 release. I didn't do any deep research in the H2 version differences beyond addressing this issue.
